feature(eas-cli): add `worker:alias|promote --prod` flag to promote existing deployments

byCedric opened the pull request:
# Why

This adds `worker:alias --prod` and supports `worker:promote --prod` (we can document either one) to promote existing deployments to production. We could split the command later, if people ask for this use case a lot.

# How

- Added `--prod(uction)` to `worker:alias`
- Added `worker:promote` alias of `worker:alias`
- Pulled both table and json output format into `src/worker/utils/logs.ts`
- Re-implemented table/json output in both `worker` and `worker:alias`
- Updated the dashboard URL from `serverless` to `hosting`

# Test Plan

- `$ eas worker`
- `$ eas worker:alias --id <deployment> --alias test-alias`
- `$ eas worker:alias --id <deployment> --alias test-alias --json --non-interactive | jq`
- `$ eas worker:promote --prod --id <deployment> --json --non-interactive | jq`

kadikraman reviewed (commented):
`eas deploy:promote`  works, but I think `eas worker:promote` has been added, I'm getting a `Error: command worker:promote not found`.

kadikraman reviewed (approved):
LGTM 🚀

byCedric commented:
> `eas deploy:promote` works, but I think `eas worker:promote` has been added, I'm getting a `Error: command worker:promote not found`.

As discussed, the command itself is called `eas worker`. Probably because it's hidden/beta, it likely disabled. But, besides that, we discussed that we are going for `eas deploy` for now.

byCedric commented:
I added a fix for the typing issue from the updates test, mostly to get this PR mergable. No idea what's going on there, but it seems that the `UpdateBranch` type was updated to include `deployments`, which caused some issues in the `mockUpdateBranch` test function. I added an existing fix for similar properties.

I'll go ahead and merge this PR, but do ping me if the fix I added is incorrect.
